internal/pkg/describe/backend_app.go

```diff
@@ -0,0 +1,215 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package describe
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+)
+
+// BackendAppDescriber retrieves information about a backend application.
+type BackendAppDescriber struct {
+	app             *archer.Application
+	enableResources bool
+
+	store            storeSvc
+	appDescriber     appDescriber
+	initAppDescriber func(string) error
+}
+
+// NewBackendAppDescriber instantiates a backend application describer.
+func NewBackendAppDescriber(project, app string) (*BackendAppDescriber, error) {
+	svc, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to store: %w", err)
+	}
+	meta, err := svc.GetApplication(project, app)
+	if err != nil {
+		return nil, err
+	}
+	opts := &BackendAppDescriber{
+		app:             meta,
+		enableResources: false,
+		store:           svc,
+	}
+	opts.initAppDescriber = func(env string) error {
+		d, err := NewAppDescriber(project, env, app)
+		if err != nil {
+			return err
+		}
+		opts.appDescriber = d
+		return nil
+	}
+	return opts, nil
+}
+
+// NewBackendAppDescriberWithResources instantiates a backend application with stack resources.
+func NewBackendAppDescriberWithResources(project, app string) (*BackendAppDescriber, error) {
+	d, err := NewBackendAppDescriber(project, app)
+	if err != nil {
+		return nil, err
+	}
+	d.enableResources = true
+	return d, nil
+}
+
+// Describe returns info of a backend application.
+func (d *BackendAppDescriber) Describe() (HumanJSONStringer, error) {
```

Comment:
Oh would you mind also adding the `URI` method? It's needed in `app_deploy`, it'd return the service discovery namespace.

Reply:
do you mean adding URI method to "BackendAppDescriber"?